boards/arm/samv7/common/scripts/flat.memory

```diff
@@ -22,10 +22,6 @@ include $(TOPDIR)/.config
 include $(TOPDIR)/tools/Config.mk
 include $(TOPDIR)/arch/arm/src/armv7-m/Toolchain.defs
 
-ifeq ($(CONFIG_CYGWIN_WINTOOL),y)
-  ARCHSCRIPT = -T "${shell cygpath -w $(BOARD_COMMON_DIR)$(DELIM)scripts$(DELIM)samv7.ld}"
-else
-  ARCHSCRIPT = -T$(BOARD_COMMON_DIR)$(DELIM)scripts$(DELIM)samv7.ld
-endif
+ARCHSCRIPT_PREPROCESS = $(BOARD_COMMON_DIR)$(DELIM)scripts$(DELIM)flat-template.ld
```

Comment:
I would like to test this change on samv7 in prior to merging. Currently there is some flexibility to auto-generate linker scripts depending on ROM loader vs MCUboot loader and I feel that we might miss this functionality with this change.

Reply:
The goal of this PR was to change how the files were generated (using dependency specification rather than at the beginning with the context rule). If something is changing then we definitely need to address that. It will be easier to view a diff of this PR after #5496 is merged

Reply:
Thank you. Let's proceed with https://github.com/apache/incubator-nuttx/pull/5496 first